chpl/chpl-service/src/main/java/gov/healthit/chpl/app/SurveillanceOversightReportWeeklyApp.java

```diff
@@ -133,47 +117,66 @@ public static void main( String[] args ) throws Exception {
         
         List<File> files = new ArrayList<File>();
         files.add(surveillanceReportFile);
-        app.getMailUtils().sendEmail(toEmail, subject, htmlMessage, files, props);
-        context.close();
+        this.getMailUtils().sendEmail(null, bccEmail, subject, htmlMessage, files, props);
 	}
 	
-	public CertifiedProductDAO getCertifiedProductDAO() {
-		return certifiedProductDAO;
-	}
-
-	public void setCertifiedProductDAO(CertifiedProductDAO certifiedProductDAO) {
-		this.certifiedProductDAO = certifiedProductDAO;
-	}
-
-	public SimpleDateFormat getTimestampFormat() {
-		return timestampFormat;
-	}
-
-	public void setTimestampFormat(SimpleDateFormat timestampFormat) {
-		this.timestampFormat = timestampFormat;
-	}
-
-	public CertifiedProductDetailsManager getCpdManager() {
-		return cpdManager;
-	}
-
-	public void setCpdManager(CertifiedProductDetailsManager cpdManager) {
-		this.cpdManager = cpdManager;
+	private void sendAcbWeeklyRecipientSubscriptionEmails(Map<CertificationBodyDTO, Set<String>> acbEmailMap, Properties props, File downloadFolder, Map<CertificationBodyDTO, CertifiedProductDownloadResponse> acbDownloadMap) throws IOException, AddressException, MessagingException {
+		String surveillanceReportFilename = null;
+        String htmlMessage = null;
+        String subject = null;
+        File surveillanceReportFile = null;
+        
+        // Send one email for each ACB to all of its subscribed recipients
+        for(Map.Entry<CertificationBodyDTO, Set<String>> entry : acbEmailMap.entrySet()){
+        	List<File> files = new ArrayList<File>();
+        	String fmtAcbName = entry.getKey().getName().replaceAll("\\W", "").toLowerCase();
+        	surveillanceReportFilename = fmtAcbName + "-" + props.getProperty("oversightEmailWeeklyFileName");
+        	surveillanceReportFile = new File(downloadFolder.getAbsolutePath() + File.separator + surveillanceReportFilename);
+            if(!surveillanceReportFile.exists()) {
+            	surveillanceReportFile.createNewFile();
+            } else {
+            	surveillanceReportFile.delete();
+            }
+            // Generate this ACB's download file  	
+            this.getPresenter().presentAsFile(surveillanceReportFile, acbDownloadMap.get(entry.getKey()));
+    		files.add(surveillanceReportFile);	
+        	
+        	subject = entry.getKey().getName() + " " + props.getProperty("oversightEmailWeeklySubject");
+        	htmlMessage = props.getProperty("oversightEmailAcbWeeklyHtmlMessage");
```

Comment:
Can you add the ACB name to this message? I think the body of the email is currently something like:
```
<h3>title</h3>
either <p>nothing</p>
or <ul>things broken</ul>
```
Can you put an `<h4>ACB Name</h4>` after the `<h3>` and before either the `<p>` or `<ul>`?